desc/magnetic_fields.py

```diff
@@ -1094,3 +1130,644 @@ def odefun(rpz, s):
+        )
+
+        assert (
+            self.surface_grid.NFP == NFP
+        ), "NFP of surface must match NFP of surface_grid"
+
+        super().__init__(
+            R_lmn, Z_lmn, modes_R, modes_Z, NFP, sym, rho, name, check_orientation
+        )
+
+    @property
+    def surface_grid(self):
+        """Grid of points to evaluate surface current at."""
+        return self._surface_grid
+
+    @surface_grid.setter
+    def surface_grid(self, new):
+        if new != self._surface_grid:
+            self._surface_grid = new
+
+    @property
+    def I(self):  # noqa: E743
+        """Net current linking the plasma and the coils toroidally."""
+        return self._I
+
+    @I.setter
+    def I(self, new):  # noqa: E743
+        assert np.isscalar(new), "I must be a scalar"
+        self._I = new
+
+    @property
+    def G(self):
+        """Net current linking the plasma and the coils poloidally."""
+        return self._G
+
+    @G.setter
+    def G(self, new):
+        assert np.isscalar(new), "G must be a scalar"
+        self._G = new
+
+    @property
+    def Phi_mn(self):
+        """Fourier coefficients describing single-valued part of potential."""
+        return self._Phi_mn
+
+    @Phi_mn.setter
+    def Phi_mn(self, new):
+        if len(new) == self.Phi_basis.num_modes:
+            self._Phi_mn = jnp.asarray(new)
+        else:
+            raise ValueError(
+                f"Phi_mn should have the same size as the basis, got {len(new)} for "
+                + f"basis with {self.Phi_basis.num_modes} modes."
+            )
+
+    @property
+    def Phi_basis(self):
+        """DoubleFourierSeries: Spectral basis for Phi."""
+        return self._Phi_basis
+
+    @property
+    def sym_Phi(self):
+        """str: Type of symmetry of periodic part of Phi (no symmetry if False)."""
+        return self._sym_Phi
+
+    def change_Phi_resolution(self, *args, **kwargs):
+        """Change the maximum poloidal and toroidal resolution for Phi."""
+        assert (
+            ((len(args) in [2, 3]) and len(kwargs) == 0)
+            or ((len(args) in [2, 3]) and len(kwargs) in [1, 2])
+            or (len(args) == 0)
+        ), (
+            "change_Phi_resolution should be called with 2 (M,N) or 3 (L,M,N) "
+            + "positional arguments or only keyword arguments."
+        )
+        L = kwargs.pop("L", None)
+        M = kwargs.pop("M", None)
+        N = kwargs.pop("N", None)
+        NFP = kwargs.pop("NFP", None)
+        sym_Phi = kwargs.pop("sym_Phi", None)
+        assert len(kwargs) == 0, "change_resolution got unexpected kwarg: {kwargs}"
+        self._NFP = NFP if NFP is not None else self.NFP
+        self._sym_Phi = sym_Phi if sym_Phi is not None else self.sym_Phi
+        if L is not None:
+            warnings.warn("Phi basis does not have radial resolution, ignoring L")
+        if len(args) == 2:
+            M, N = args
+        elif len(args) == 3:
+            L, M, N = args
```

Comment:
Most of this was to account for surfaces being either L/M or M/N, but in this case we know its always M/N so we can probably get rid of most of this.